tests/copp/test_copp.py

```diff
@@ -313,3 +380,50 @@ def _teardown_multi_asic_proxy(dut, creds, test_params, tbinfo):
     ns_ip = dut.shell("sudo ip -n {} -4 -o addr show eth0".format(test_params.nn_target_namespace) + " | awk '{print $4}' | cut -d'/' -f1")["stdout"]
     dut.command("sudo iptables -t nat -D PREROUTING -p tcp --dport 10900 -j DNAT --to-destination {}".format(ns_ip))
     dut.command("sudo ip -n {} rule delete from {} to {} pref 3 lookup default".format(test_params.nn_target_namespace, ns_ip, tbinfo["ptf_ip"]))
+
+def _check_trap_counter_enabled(duthost, trap_type):
+    lines = duthost.command('show flowcnt-trap stats')['stdout']
+    return trap_type in lines
+
+def _collect_counter_rate(duthost, trap_type, actual_rate):
+    rate_values = []
+    # Wait up to _SEND_DURATION + 5 seconds for PTF to stop sending packet,
+    # as it might take some time for PTF to initialize itself
+    end_time = time.time() + _SEND_DURATION + 5
+    packets = None
+    while time.time() < end_time:
```

Comment:
Why not just wait for "_SEND_DURATION + 5" and then check? I suppose PTF will only send out all the packets after the time "_SEND_DURATION + 5"